api/swagger-batch.yaml

```diff
@@ -0,0 +1,310 @@
+
+    EnsemblingJobSpec:
+      description: Ensembling job Specification
+      properties:
+        source:
+          $ref: "#/components/schemas/EnsemblingJobSource"
+        predictions:
+          type: "array"
+          items:
+            $ref: "#/components/schemas/EnsemblingJobPredictionSource"
+        ensembler:
+          $ref: "#/components/schemas/EnsemblingJobEnsemblerSpec"
+
+    EnsemblingJobSource:
+      type: "object"
+      properties:
+        dataset:
+          $ref: "#/components/schemas/Dataset"
+        join_on:
+          type: "array"
+          items:
+            type: "string"
+
+    Dataset:
+      type: "object"
+      required:
+        - "type"
+      properties:
+        type:
+          $ref: "#/components/schemas/DatasetType"
+      discriminator:
+        propertyName: "type"
+
+    DatasetType:
+      type: "string"
+      enum:
+        - "BQ"
+
+    BigQueryDataset:
+      allOf:
+        - $ref: "#/components/schemas/Dataset"
+        - type: object
+          properties:
+            bq_config:
+              $ref: "#/components/schemas/BigQueryDatasetConfig"
+
+    BigQueryDatasetConfig:
+      properties:
+        table:
+          type: "string"
+          pattern: '^[a-z][a-z0-9-]+\.\w+([_]?\w)+\.\w+([_]?\w)+$'
+        features:
+          type: "array"
+          items:
+            type: "string"
+        query:
+          type: "string"
+        options:
+          type: "object"
+          additionalProperties:
+            type: "string"
+
+    EnsemblingJobPredictionSource:
+      type: "object"
+      allOf:
+        - $ref: "#/components/schemas/EnsemblingJobSource"
+        - type: "object"
+          properties:
+            columns:
+              type: "array"
+              items:
+                type: "string"
+
+    EnsemblingJobEnsemblerSpec:
+      type: "object"
+      properties:
+        uri:
+          type: "string"
+          format: "uri"
+        result:
+          type: "object"
+          properties:
+            type:
+              $ref: "#/components/schemas/EnsemblingJobResultType"
+            item_type:
+              $ref: "#/components/schemas/EnsemblingJobResultType"
+            column_name:
+              type: "string"
+
+    EnsemblingJobResultType:
+      type: "string"
+      enum:
+        - DOUBLE
+        - FLOAT
+        - INTEGER
+        - LONG
+        - STRING
+        - ARRAY
+
+    PaginatedResults:
+      type: "object"
+      properties:
+        paging:
+          type: "object"
+          properties:
+            total:
+              type: "integer"
+              format: "int32"
+              description: "Total number of results matching the query criteria"
+            page:
+              type: "integer"
+              format: "int32"
+              description: "Number of the current page"
+            pages:
+              type: "integer"
+              format: "int32"
+              description: "Total number of pages"
+
```

Comment:
Since a number of these data structures are meant to match the [Proto](https://github.com/gojek/turing/blob/main/engines/batch-ensembler/api/proto/v1/batch_ensembling_job.proto), I'm wondering if it would make sense to auto-generate them using a library like [this](https://github.com/google/gnostic) and reference them by splitting the OpenAPI specs.

Reply:
Interesting suggestion! :) 
When I defined the original [proto](https://github.com/gojek/turing/blob/main/engines/batch-ensembler/api/proto/v1/batch_ensembling_job.proto), I spent some time thinking about how to reduce this repetitive effort of maintaining both openapi spec and proto spec, but haven't decided how exactly to approach it. In my opinion, there is no much benefits of maintaining the proto spec, it can probably be replaced with openapi spec alone. But since ensembling job is in scope of `engines/batch-ensembler`, and openapi spec belongs to `api`, it made it hard to merge them easily. 
Maybe in the future I'll just move openapi specs into the root level of the project, remove the proto and use [openapi-generator](https://github.com/OpenAPITools/openapi-generator) to generate python/golang code of these data structures.

Reply:
Just to make it clear, I don't plan to work or on this idea in this PR :)